Add ability to change Discovery Prefix via WebUI or via mosquitto_pub

puterboy opened the pull request:
## Description:
Changes made as discussed in https://github.com/1technophile/OpenMQTTGateway/issues/1996
Specifically:
1. Declaration `discovery_Topic` deprecated (but still working) and changed to `discovery_Prefix` to avoid confusion with `discovery_topic`
2. New *variable* `discovery_prefix` introduced that defaults on new installs to discovery_Prefix (or discovery_Topic if present), is stored in NVS, and can be changed via WebUI  (under `Configure MQTT`) or via mosquitto_pub, using something like:
```
mosquitto_pub -t "home/<GATEWAY_NAME>/commands/MQTTtoSYS/config" -m '{  "discovery_prefix": "<NEW_DISCOVERY_PREFIX" }'
```

I tested it using an esp32 with `esp32dev-ble` and the functionality works as described.
Let me know if you are ok with the PR.

## Checklist:
  - [X ] The pull request is done against the latest development branch
  - [X ] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] I accept the [DCO](https://github.com/1technophile/OpenMQTTGateway/blob/development/docs/participate/development.md#developer-certificate-of-origin).

puterboy committed:
Update main.ino with discovery_prefix

Add discovery_prefix

puterboy committed:
Update config_mqttDiscovery.h

puterboy committed:
Update config_mqttDiscovery.h

puterboy committed:
Update ZmqttDiscovery.ino with discovery_prefix instead of discovery_Topic

puterboy committed:
Update config_WebContent.h - Update with discovery_prefix

puterboy committed:
Update ZwebUI.ino to include discovery_prefix

puterboy committed:
Update ZwebUI.ino

puterboy committed:
Update main.ino

puterboy committed:
Update main.ino

1technophile commented on main/main.ino:
```diff
@@ -1332,6 +1332,7 @@ void setup() {
   Log.trace(F("OpenMQTTGateway ip: %s" CR), WiFi.localIP().toString().c_str());
   Log.trace(F("OpenMQTTGateway index %d" CR), cnt_index);
   Log.trace(F("OpenMQTTGateway mqtt topic: %s" CR), mqtt_topic);
+  Log.trace(F("OpenMQTTGateway mqtt topic: %s" CR), discovery_prefix);
```
OpenMQTTGateway mqtt discovery prefix

1technophile commented:
Thanks, could you update the documentation also.

The build ko for the AWS environment is because some environments do not use discovery. Your change needs to be included when the macro ZmqttDiscovery is defined.

Finally check the box for the DCO in the PR and we will be good

puterboy committed:
Update main.ino - added #ifdef ZmqttDiscovery

puterboy committed:
Update config_WebContent.h - added #ifdef ZmqttDiscovery

puterboy committed:
Update ZwebUI.ino - added ZmqttDiscovery

puterboy committed:
Update gateway.md - Added discovery_prefix

puterboy committed:
Merge branch '1technophile:development' into puterboy-patch-1

puterboy commented:
- Corrected the typo where mqtt_topic was used instead of discovery_prefix
- Added #ifdef ZmqttDiscovery in all *new* places that reference discovery_prefix (note if a line of code previously referenced discovery_Topic and was not conditioned before then I left it as is now)
- Added discovery_prefix key to the mosquito_pub documentation (Note the documentation doesn't address setting "MQTT Base Topic" except in initial startup where "Discovery Prefix" is not relevant -- so I didn't know where or if I should document "Discovery Prefix" for the regular WebUI

puterboy committed:
Update config_WebContent.h - typo fixed

puterboy committed:
Update config_mqttDiscovery.h - Changed from parameters_size to mqtt_topic_max_size

puterboy committed:
Update ZwebUI.ino

puterboy committed:
Revert "Update ZwebUI.ino"

This reverts commit 2ed71b1f00fbdc2583e929856cd6205618b1e2ee.

puterboy committed:
Revert "Update config_mqttDiscovery.h - Changed from parameters_size to mqtt_topic_max_size"

This reverts commit fa0ba043be99982ded5d336ae08bbeb360184a6f.

puterboy commented:
Not sure what the lint formatting issues are - I tried to copy how you spaced and indented macro if/else/end statements.

puterboy committed:
Update main.ino corrected typo where mqtt topic instead of mqtt discovery prefix

1technophile commented:
So as to format automatically your document you have to add the "clang-Format" extension to VSCode, once done, you can format the code by doing a right click into the code file window and clicking "Format document".

puterboy commented:
I was actually doing the editing in the web version of github – not VS.

Do you have the ability to fix the formatting on my patch?

 

From: Florian ***@***.***> 
Sent: Friday, August 16, 2024 6:00 PM
To: 1technophile/OpenMQTTGateway ***@***.***>
Cc: puterboy ***@***.***>; Author ***@***.***>
Subject: Re: [1technophile/OpenMQTTGateway] Add ability to change Discovery Prefix via WebUI or via mosquitto_pub (PR #2007)

 

So as to format automatically your document you have to add the "clang-Format" extension to VSCode, once done, you can format the code by doing a right click into the code file window and clicking "Format document".

—
Reply to this email directly, view it on GitHub <https://github.com/1technophile/OpenMQTTGateway/pull/2007#issuecomment-2294366069> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAPGZASM3YHZMXS7C3KQTGTZRZY4LAVCNFSM6AAAAABMTOJCJKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEOJUGM3DMMBWHE> .
You are receiving this because you authored the thread.Message ID: ***@***.***>

1technophile commented:
Yes sure

puterboy commented:
Thanks!

1technophile committed:
Fix format
